internal/maintenance/rescuer.go

```diff
@@ -146,10 +146,15 @@ func (s *Rescuer) Start(ctx context.Context) error {
 }
 
 type rescuerRunOnceResult struct {
+	NumJobsCancelled int64
 	NumJobsDiscarded int64
 	NumJobsRetried   int64
 }
 
+type metadataWithCancelledByQuery struct {
+	CancelledByQuery bool `json:"cancelled_by_query"`
```

Comment:
What do you mean exactly by "cancelled by query"? Does that just mean that it was cancelled and cancellation happens by a query so that it was cancelled by a query?

Reply:
It's a little tricky because of the inherent raciness of this operation. The cancel query itself is guaranteed to be completed here, but the job isn't actually cancelled yet at the moment this is set because that's an asynchronous operation which requires the producer to receive the signal, act on it, and the job to then exit. And there are the edge cases where the job completes or gets discarded before the cancel signal is received.

Perhaps `cancel_attempted_at` or `cancel_initiated_at` is a suitable name for this attr and/or a column?